testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/single/web/NonHaWebSessionPersistenceTestCase.java

```diff
@@ -99,8 +103,11 @@ public void testSessionPersistence(@ArquillianResource(SimpleServlet.class) @Ope
             Assert.assertFalse(Boolean.valueOf(response.getFirstHeader("serialized").getValue()));
             response.getEntity().getContent().close();
 
-            stop(CONTAINER_SINGLE);
-            start(CONTAINER_SINGLE);
+            NodeUtil.stop(this.controller, CONTAINER_SINGLE);
+            NodeUtil.start(this.controller, CONTAINER_SINGLE);
+            if (Boolean.getBoolean("ts.bootable")) {
```

Comment:
Hi @yersan - I might be missing something here, but IIUC the deployment is operated by Arquillian only when `ts.bootable` is set. What happens for Galleon - i.e. when `ts.layers` is set?

Reply:
Hi @fabiobrz This test requires a server restart to later validate the session information is correctly recovered. This test is used for a bootable JAR, Galleon trimmed or bare metal server. The deployment is always managed by the test itself, I mean Arquillian will not automatically deploy the application, we deploy it manually on @Before and @After methods, but the deployment survives to a server restart when we are not in the Bootable JAR scenario.

When we are testing the Bootable JAR we need to redeploy it after stopping the server, hence this `if`.

Reply:
Thanks @yersan you’re right, the jar must be redeployed to the hollow bootable jar server.